doc/examples/example_asymm_crypto.md

```diff
@@ -476,3 +476,101 @@ mococrw::EdDSASignaturePrivateKeyCtx ctx(key);
+
+CRYSTALS-Dilithium is a lattice-based signature scheme. There are no known quantum computer alogrithms which are able to break dilithium.
+
+For further information regarding dilithium please refer to https://pq-crystals.org/dilithium/index.shtml .
+
+## Note on hashing algorithms
+Dilithium differs compared to the other algorithms. The other signing-/verification-algorithms described here expect a hashing algorithm when verifyMessage(...) or signMessage(...) is used. Dilithium does not.
+
+The reason for this behavior is that the dilithium signature algorithm can take arbitrary length inputs and internally digests them to obtain a fixed-length signature. As a consequence, messages may be directly passed into it or a hash algorithm of choice (e.g., SHA3-512) can be used to digest the message and then pass this digest into the verifyMessage interface here. This can be an implementation benefit as the dilithium interface doesn't support streaming for large amounts of data but only to pass all data in one go.
+
+## Note on DER format
+For key exchange the format is expected to be according to RFC5208-PrivateKeyInfo and RFC5280-SubjecPublicKeyInfo. These standards define the private and public key as ASN.1 octet or bit string.
+As there is no ASN.1 format for the de-/serialisation of dilithium keys the one which is used in MoCOCrW looks like the following:
+* Private Key: embedded in the PrivateKeyInfo:PrivateKey octet string ASN.1 field as defined in [RFC5208 Section 5](https://www.rfc-editor.org/rfc/rfc5208#section-5)
+```
+ASN1_OCTET_STRING privKey
+ASN1_OCTET_STRING pubKey
+ASN1_INTEGER dilithiumParameterSet
+ASN1_BOOLEAN bool1
+ASN1_BOOLEAN bool2
+```
+* Public Key: embedded in the SubjectPublicKeyInfo:subjectPublicKey bit string ASN.1 field as defined in [RFC5280 Section 4.1](https://www.rfc-editor.org/rfc/rfc5280#section-4.1)):
+```
+ASN1_OCTET_STRING pubKey
+ASN1_INTEGER dilithiumParameterSet
+ASN1_BOOLEAN bool1
+ASN1_BOOLEAN bool2
+```
+
+The format might change once a standard for the key format is published.
+
+## Note on future (in)compatibility
+
+The current implementation for RSA, ECC, EcDSA, ... is EVP-centric. EVP is an abstract internal representation of keys in openssl.
+
+As the changes required for getting dilithium into the current EVP-centric implementation would have broken the API we decided to "clone" the API of the existing implementation and make the dilithium implementation as similar as possible.
+
+Once openssl supports dilithium, this shall allow us to create aliases for the dilithium classes. These aliases will then point to the EVP-centric classes. This will break the API.
+
+To be as little affected as possible by the API break the use of `auto` for dilithium classes is recommended and might then only require a recompilation but no code changes.
```

Comment:
```suggestion
To be as little affected as possible by the API break, the use of `auto` for dilithium classes is recommended. It might then only require a recompilation, but no code changes.
```